transaction.go

```diff
@@ -0,0 +1,44 @@
+package datastore
+
+// basicTransaction implements the transaction interface for datastores who do
+// not have any sort of underlying transactional support
+type basicTransaction struct {
+	puts    map[Key]interface{}
+	deletes map[Key]struct{}
+
+	target Datastore
+}
+
+func NewBasicTransaction(ds Datastore) Transaction {
+	return &basicTransaction{
+		puts:    make(map[Key]interface{}),
+		deletes: make(map[Key]struct{}),
+		target:  ds,
+	}
+}
```

Comment:
it does, it makes it clear that this is not transactional and is not atomic and people should not expect it to be.